### DISCUSSION TITLE
Does LaplacianPositional encoding matrix correspond to the same order as the node feature matrix?

### DISCUSSION THREAD

**Discussion by amjass12** [pyg-team/pytorch_geometric#8081] [Q&A] [answered]
Hi all!

I am running Laplacianeigenvector positional encoding on my torch geometric dataset, however,  I am unsure about whether the PE matrix returned is in the same order as in the node features in the torch dataset. Please can somebod confirm? many thanks. 

code:
```
from torch_geometric.transforms import AddLaplacianEigenvectorPE
pos_enc = AddLaplacianEigenvectorPE(k=50)
pos_enc(graph_dataset)

```
now 'laplacian_encoding_pe' matrix is added as expected, but is the order of the matrix the same as the node features (data.x)?

so is PE[0] the positional encoding for node 0 (as in data.x[0]) and so on

Many thanks!!

**Comment by rusty1s** [comment, marked as answer]
Yes, this should work. I added a test to confirm: https://github.com/pyg-team/pytorch_geometric/pull/8087

**Comment by amjass12** [comment]
Hi @rusty1s ,

thank you very much for confirming! would it be possible to implement an option for padding if the selected `k` has more dimensions than eigenvectors?

thank you!

**Reply by rusty1s** [reply]
Yes, that is possible. Do you wanna contribute this?

**Reply by amjass12** [reply]
for sure, will start working on this as soon as i have time! thanks again!
